scripts/bugzilla_attach.py

```diff
@@ -0,0 +1,353 @@
+                "--image=" + image, "--dest-dir=" + logfolder],
+                stdout=out_file, check=True)
+        except subprocess.CalledProcessError:
+            print("Error in the execution of must-gather:")
+            exit(1)
+
+
+def trim_logs(logfolder, num_seconds):
+    for subdir, _, files in os.walk(logfolder):
+        for file in files:
+            if file == "must-gather.log": #Ignore the log made by capturing the output of must-gather
+                continue
+            if ".log" in file:
+                trim_file_by_time(os.path.join(subdir, file), num_seconds, PODLOG_TIMESTAMP_REGEX, PODLOG_TIMESTAMP_FORMAT)
+            if "kubelet" in file or "NetworkManager" in file:
+                trim_file_by_time(os.path.join(subdir, file), num_seconds, NODELOG_TIMESTAMP_REGEX, NODELOG_TIMESTAMP_FORMAT)
+
+def trim_node_logs(logfolder, num_seconds):
+    print("Trimming node logs")
+    for node_name in os.listdir(logfolder + "nodes/"):
+        trim_file_by_time(os.path.join(logfolder, "nodes", node_name, node_name + "_logs_kubelet"), num_seconds, NODELOG_TIMESTAMP_REGEX, NODELOG_TIMESTAMP_FORMAT)
+        trim_file_by_time(os.path.join(logfolder, "nodes", node_name, node_name + "_logs_NetworkManager"), num_seconds, NODELOG_TIMESTAMP_REGEX, NODELOG_TIMESTAMP_FORMAT)
+
+def trim_pod_logs(logfolder, num_seconds):
+    print("Trimming pod logs")
+    for namespace in os.listdir(logfolder + "namespaces/"):
+        pod_folder = os.path.join(logfolder, "namespaces", namespace, "pods")
+        if not os.path.exists(pod_folder):
+            continue
+        for subdir, _, files in os.walk(pod_folder):
+            for file in files:
+                if ".log" in file:
+                    trim_file_by_time(os.path.join(subdir, file), num_seconds, PODLOG_TIMESTAMP_REGEX, PODLOG_TIMESTAMP_FORMAT)
+
+
+def try_parse_int(value):
+    """Tries to parse the value as an int"""
+    try:
+        return int(value), True
+    except ValueError:
+        return value, False
+
+
+
+def send_data(bug_id, file_name, file_data, comment, authentication):
+    """Sends the data to Bugzilla with the relevant information"""
+    url = BUGZILLA_URL + '/rest/bug/%s/attachment' % bug_id
+    data = {
+        **authentication,
+        "ids": [bug_id],
+        "comment": comment,
+        "summary": "Result from must-gather command",
+        "content_type": "application/gzip",
+        "file_name": file_name,
+        "data": file_data
+    }
+    return requests.post(url, json=data, headers=HEADERS)
+
+
+def check_bug_exists(bug_id):
+    """Checks whether the bug exists in Bugzilla"""
+    url = BUGZILLA_URL + '/rest/bug/%s' % bug_id
+    return "error" not in requests.get(url).json()
+
+
+def trim_file_by_time(filename, num_seconds, timestamp_regex, timestamp_format):
+    """Uses a binary search to locate where in the file to trim"""
+    with open(filename, "r+") as file:
+        lines = [line.rstrip('\n') for line in file]
```

Comment:
Why not to reverse the lines and start from the end to the time we wanted?

Reply:
The binary search is much faster than traversing from the end, since traversing from the end would need comparison of each line (which could be thousands) whereas even on large files I've had fewer than 25 comparisons.

Reply:
OK